plugins/modules/infra_join_token.py

```diff
@@ -0,0 +1,226 @@
+            # if existing is None, then it is a create operation
+            return True
+
+        return self.is_changed(self.existing.model_dump(by_alias=True, exclude_none=True), self.payload_params)
+
+    def find(self):
+        if self.params["id"] is not None:
+            try:
+                resp = UIJoinTokenApi(self.client).read(self.params["id"])
+                return resp.result
+            except NotFoundException as e:
+                if self.params["state"] == "revoked":
+                    return None
+                raise e
+        else:
+            filter = f"name=='{self.params['name']}'"
+            resp = UIJoinTokenApi(self.client).list(filter=filter)
+
+            # If no results, set results to empty list
+            if not resp.results:
+                resp.results = []
+
+            if len(resp.results) == 1:
+                return resp.results[0]
+            if len(resp.results) > 1:
+                self.fail_json(msg=f"Found multiple View: {resp.results}")
+            if len(resp.results) == 0:
+                return None
+
+    def create(self):
+        if self.check_mode:
+            return None
+
+        resp = UIJoinTokenApi(self.client).create(body=self.payload)
+        return resp.result.model_dump(by_alias=True, exclude_none=True)
+
+    def update(self):
+        if self.check_mode:
+            return None
+
+        update_body = self.payload
+        update_body = self.validate_readonly_on_update(self.existing, update_body, ["name", "description"])
+
+        resp = UIJoinTokenApi(self.client).update(id=self.existing.id, body=update_body)
+        return resp.result.model_dump(by_alias=True, exclude_none=True)
+
+    def delete(self):
+        if self.check_mode:
+            return
+
+        UIJoinTokenApi(self.client).delete(self.existing.id)
+
+    def run_command(self):
+        result = dict(changed=False, object={}, id=None)
+
+        # based on the state that is passed in, we will execute the appropriate
+        # functions
+        try:
+            self.existing = self.find()
+            item = {}
+            if self.params["state"] == "present" and self.existing is None:
+                item = self.create()
+                result["changed"] = True
+                result["msg"] = "JoinToken created"
+            elif self.params["state"] == "present" and self.existing is not None:
+                if self.payload_changed():
+                    item = self.update()
+                    result["changed"] = True
+                    result["msg"] = "JoinToken updated"
+            elif self.params["state"] == "revoked" and self.existing is not None and self.existing.status != "REVOKED":
+                self.delete()
+                result["changed"] = True
+                result["msg"] = "JoinToken Revoked"
+            elif self.params["state"] == "revoked" and self.existing is not None and self.existing.status == "REVOKED":
+                self.delete()
```

Comment:
do you have to call delete again if it is already revoked? 

Also, it will be interesting to see how the module behaves when a task is made to revoke a token that does not exist.

Reply:
No , delete call removed there.

When tried to delete a token that has never been created , the filter fails as it cant find the token and the task passes as no action is otherwise performed.

Reply:
Ok. I also wonder what is the value of "changed". Based on the code, I feel that we don't set it in that scenario, but it might be the case with other modules as well when we try to do an "absent" task for a non existing resource. 

However in this case, I also wonder if "revoked" task should throw an error if not found, since it is not equivalent to "absent".